packages/cozy-konnector-libs/src/libs/saveIdentity.js

```diff
@@ -145,9 +140,57 @@ function formatPhone(phone) {
     if (element.number) {
       element.number = element.number.replace(/[^\d.+]/g, '')
       phone[phone.indexOf(element)] = element
+    } else {
+      log('warn', 'There is no number in phone object')
     }
   }
   return phone
 }
 
-module.exports = saveIdentity
+function formatIdentityContact(contactToFormat) {
+  const contact = { ...contactToFormat }
+  if (contact.phone) {
+    if (!Array.isArray(contact.phone)) {
+      log(
+        'warn',
+        'formatIdentityContact Phone is not an array, you should fix it'
+      )
+      contact.phone = [
```

Comment:
If I may, here we usually use the following structure :

```
const phone = [
  {
   number : 0000000000,
   type : 'mobile'
  }
]
```
The type is determine with the first two numbers of the phoneNumber.
Maybe we could had it here too, to prevent further issues ?

Something like : 

```
if (contact.phone) {
    if (!Array.isArray(contact.phone)) {
      log(
        'warn',
        'formatIdentityContact Phone is not an array, you should fix it'
      )
      let type
      if(contact.phone.startsWith('06') || contact.phone.startsWith('07'){
        type = 'mobile
      }else{
        type = 'home'
      }
      contact.phone = [
      {
          number: contact.phone,
          type
        }
      ]
    }
    contact.phone = formatPhone(contact.phone)
  
```
What do you think ?

Reply:
By reading https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.contacts.md#iocozycontacts 

`type` is not mandatory. So I don't want to add black magic here. You can have a `work` mobile phone. This is something to handle at the konnector level. Here we don't know anything.

Reply:
Alright then